### PR TITLE
Osrm compat roundabouts

### PR DESCRIPTION
**Pull request by gknisely**
added logic to support circulars. They will be considered roundabouts in valhalla.
walk should be walking. 
Added logic for roundabouts. Use name for roundabout if available.

### CODE REVIEW

File: src/tyr/route_serializer.cc [hunk 1]
```diff
@@ -488,7 +488,8 @@ namespace {
     json::MapPtr osrm_maneuver(const valhalla::odin::TripDirections::Maneuver& maneuver,
                 std::list<odin::TripPath>::const_iterator path_leg,
                 const PointLL& man_ll, const bool depart, const bool arrive,
-                const uint32_t count, const std::string mode, const std::string prev_mode) {
+                const uint32_t count, const std::string mode, const std::string prev_mode,
```
Comment (dnesbitt61) [inline src/tyr/route_serializer.cc]: pass strings by reference

File: src/tyr/route_serializer.cc [hunk 2]
```diff
@@ -526,7 +527,8 @@ namespace {
         const auto& current_edge = path_leg->node(idx).edge();
         bool new_name = maneuver.type() == odin::TripDirections_Maneuver_Type_kContinue ||
                         maneuver.type() == odin::TripDirections_Maneuver_Type_kBecomes;
-        bool roundabout = false;
+        bool roundabout = maneuver.type() == odin::TripDirections_Maneuver_Type_kRoundaboutEnter ||
+                          maneuver.type() == odin::TripDirections_Maneuver_Type_kRoundaboutExit;
```
Comment (dnesbitt61) [inline src/tyr/route_serializer.cc]: is this roundabout boolean used anymore?

File: src/tyr/route_serializer.cc [hunk 3]
```diff
@@ -541,8 +543,18 @@ namespace {
           maneuver_type = ramp_type(prior_edge, idx, path_leg);
         } else if (new_name) {
           maneuver_type = "new name";
-        } else if (roundabout) {
-          maneuver_type = "roundabout";
+        } else if (maneuver.type() == odin::TripDirections_Maneuver_Type_kRoundaboutEnter) {
```
Comment (dnesbitt61) [inline src/tyr/route_serializer.cc]: should roundabout come before new_name? maybe it should be split out like arrive and depart?

File: src/tyr/route_serializer.cc (hunk 1 above)
Comment (gknisely) [reply src/tyr/route_serializer.cc]: Thanks.  Missed that.

File: src/tyr/route_serializer.cc (hunk 2 above)
Comment (gknisely) [reply src/tyr/route_serializer.cc]: yeah not really needed.

File: src/tyr/route_serializer.cc (hunk 3 above)
Comment (gknisely) [reply src/tyr/route_serializer.cc]: sure

**Pull request merged**